### PR TITLE
User lockouts

### PR DESCRIPTION
**Pull request by Alroniks**
https://oat-sa.atlassian.net/browse/TAO-6017

### CODE REVIEW

File: actions/class.Main.php [hunk 1]
```diff
@@ -182,12 +182,12 @@ public function login()
...
 
-                            $seconds = $endTime->getTimestamp() - $reference->getTimestamp();
+                            $diffInSeconds = $endTime->getTimestamp() - $reference->getTimestamp();
 
                             $msg .= __('Please try in %s.',
-                                $seconds > 60
+                                $diffInSeconds > 60
                                     ? tao_helpers_Date::displayInterval($statusDetails['remaining'], tao_helpers_Date::FORMAT_INTERVAL_LONG)
-                                    : $seconds . ' seconds'
+                                    : $diffInSeconds . '_' . ($diffInSeconds == 1 ? __('second') : __('seconds'))
```
Comment (ssipasseuth) [inline actions/class.Main.php]: the `_` should go I suppose

**Comment by ssipasseuth** [comment]
Issue on point 1 has been considered as a bug https://oat-sa.atlassian.net/browse/TAO-6099

File: actions/class.Main.php [hunk 2]
```diff
@@ -138,49 +144,84 @@ public function login()
...
 
-				if($success){
-				    \common_Logger::i("Successful login of user '" . $myForm->getValue('login') . "'.");
+                /** @var UserLocksService $userLocksService */
+                $userLocksService = $this->getServiceLocator()->get(UserLocksService::SERVICE_ID);
+                /** @var EventManager $eventManager */
+                $eventManager = $this->getServiceLocator()->get(EventManager::SERVICE_ID);
 
-                    $eventManager->trigger(new LoginSucceedEvent($myForm->getValue('login')));
+                if ($userLocksService->isLocked($form->getValue('login'))) {
```
Comment (jbout) [inline actions/class.Main.php]: Is there no way to disable this?

File: views/js/controller/users/index.js [hunk 3]
```diff
@@ -2,9 +2,32 @@
...
 
+    var runUserAction = function runUserAction(uri, action, confirmMessage) {
+        var tokenName = module.config().xsrfTokenName;
+        var data = {};
+
+        data.uri = uri;
+        data[tokenName] = $.cookie(tokenName);
+
+        dialogConfirm(confirmMessage, function () {
+            $.ajax({
```
Comment (ssipasseuth) [inline views/js/controller/users/index.js]: the new best practice requires using `core/dataProvider/request` to make ajax call to have a contralized wait to handle requests: https://github.com/oat-sa/tao-core/blob/216f8a15ba129ae2b2243571577af6628f3ca14e/views/js/core/dataProvider/request.js

Comment (ssipasseuth) [reply views/js/controller/users/index.js]: but since it is a copy/paste of an old, it can pass

**Comment by ssipasseuth** [comment]
front end code styles commented

**Pull request merged**